private/tools/java/com/github/bazelbuild/rules_jvm_external/junit5/ActualRunner.java

```diff
@@ -0,0 +1,102 @@
+package com.github.bazelbuild.rules_jvm_external.junit5;
+
+import org.junit.platform.engine.discovery.DiscoverySelectors;
+import org.junit.platform.launcher.LauncherConstants;
+import org.junit.platform.launcher.core.LauncherConfig;
+import org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder;
+import org.junit.platform.launcher.core.LauncherFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import static java.nio.file.StandardOpenOption.DELETE_ON_CLOSE;
+import static java.nio.file.StandardOpenOption.TRUNCATE_EXISTING;
+import static java.nio.file.StandardOpenOption.WRITE;
+import static org.junit.platform.launcher.EngineFilter.includeEngines;
+
+public class ActualRunner implements RunsTest {
+
+  @Override
+  public boolean run(String testClassName) {
+    var out = System.getenv("XML_OUTPUT_FILE");
+    Path xmlOut;
+    try {
+      xmlOut = out != null ? Paths.get(out) : Files.createTempFile("test", ".xml");
+      Files.createDirectories(xmlOut.getParent());
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+
+    try (var bazelJunitXml = new BazelJUnitOutputListener(xmlOut)) {
+      var summary = new CommandLineSummary();
+
+      LauncherConfig config = LauncherConfig.builder()
+        .addTestExecutionListeners(bazelJunitXml, summary)
+        .build();
+
+      var classSelector = DiscoverySelectors.selectClass(testClassName);
+
+      var request = LauncherDiscoveryRequestBuilder.request()
+        .selectors(List.of(classSelector))
+        .filters(includeEngines(
+          "junit-jupiter",
+          "junit-vintage"))
```

Comment:
It should be possible to customize these since there are a lot of other engines out there.

Reply:
Agreed. Do you have any suggestions on the best way to do this?

Reply:
I'm not a Bazel user, just a JUnit maintainer. 🙂